source/main.py

```diff
@@ -0,0 +1,253 @@
+    return rendered
+
+
+def render_table(term: Terminal, puzzles: List[dict], entries: int, index: int = 0) -> str:
+    """Render table
+
+    Args:
+        term (Terminal): blessed.Terminal object
+        puzzles (List[dict]): list of puzzles and their highscores
+        entries (int): length of 'puzzles'
+        index (int): currently selected
+
+    Returns:
+        str: rendered string
+    """
+    rendered = ''
+
+    rendered += (
+        '    ' + set_string_length('Puzzle Name', term.width - 52)
+        + '   ' + 'Player Name     '
+        + '  ' + 'Highscore                  '
+    ) + '\n\n'
+
+    for i in range(entries):
+        puzzle = puzzles[i]
+
+        line = '    '
+        line += set_string_length(puzzle['Puzzle'], term.width - 52) + '   '
+        line += set_string_length(puzzle['Name'], 16) + '  '
+        line += set_string_length(str(puzzle['Time']) + ' Seconds / ' + str(puzzle['Moves']) + ' Moves', 27)
+        rendered += term.black_on_white(line) if index == i else line
+        rendered += '\n'
+
+    rendered += term.move_y(term.height - 3)
+    rendered += '    ' + (term.black_on_white if index == entries else str)('Add new')
+    return rendered
+
+
+def render_file_explorer(term: Terminal, current_dir: str, files: List[str], dirs: List[str], index: int = 0) -> str:
+    """Render file explorer
+
+    Args:
+        term (Terminal): blessed.Terminal object
+        current_dir (str): currently selected directory
+        files (List[str]): list of files
+        dirs (List[str]): list of directories
+        index (int, optional): index of selected file/dir in parent directory. Defaults to 0.
+
+    Returns:
+        str: [description]
+    """
+    rendered = ''
+    rendered += term.black_on_white(set_string_length('    ' + current_dir, term.width)) + '\n\n'
+
+    rendered += '\n'.join([
+        (term.black_on_blue if index == i else term.blue)
+        (set_string_length(dirs[i], term.width))
+        for i in range(len(dirs))
+    ]) + '\n'
+
+    rendered += '\n'.join([
+        (term.black_on_white if index - len(dirs) == i else str)
+        (set_string_length(files[i], term.width))
+        for i in range(len(files))
+    ]) + '\n'
+
+    return rendered
+
+
+def select_new_puzzle(term: Terminal, current_dir: str = os.getcwd()) -> str:
+    """Select image from local files
+
+    Args:
+        term (Terminal): blessed.Terminal objet
+        current_dir (str, optional): directory currently selected. Defaults to os.getcwd().
+
+    Returns:
+        str: image path
+    """
+    print(term.clear)
+    print(term.home)
+
+    dirs = ['    ../']
+    files = []
+
+    for file in os.listdir(current_dir):
+        if file.endswith(SUPPORTED_FILE_TYPES) and os.path.isfile(os.path.join(current_dir, file)):
+            files.append('    ' + file)
+        elif os.path.isdir(os.path.join(current_dir, file)):
+            dirs.append('    ' + file + '/')
```

Comment:
I don't understand the spaces in line 159, 164 and 166. Are they positioning/line identation?